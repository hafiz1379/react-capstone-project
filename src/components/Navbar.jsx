import { useState, useEffect } from 'react';
import { useNavigate, useLocation } from 'react-router-dom';

const Navbar = () => {
  const navigate = useNavigate();
  const location = useLocation();
  const [showButton, setShowButton] = useState(false);
  const { pathname } = location;
  let title;

  useEffect(() => {
    setShowButton(location.pathname !== '/');
  }, [location.pathname]);

  const handleGoBack = () => {
    navigate(-1);
  };
  if (pathname === '/') {
    title = 'Air Pollution app';
  } else if (pathname.startsWith('/') && pathname.split('/').length === 2) {
    title = 'Stats';
  }

  return (
    <header className="flex bg-blue_header p-2 justify-between">
      { showButton && (
      <button type="button" className="flex items-center text-xl" onClick={handleGoBack}>
        <img src="https://api.iconify.design/ic:sharp-arrow-back-ios.svg?color=%23ffffff" alt="back arrow" />
        2023
      </button>
      )}
      <h1 className="font-normal">{title}</h1>
      <div className="flex gap-3">
        <img className="h-6" src="https://api.iconify.design/carbon:microphone-filled.svg?color=%23ffffff" alt="speak" />
        <img className="h-6" src="https://api.iconify.design/clarity:settings-solid.svg?color=%23ffffff" alt="settings" />
      </div>
    </header>
  );
};
export default Navbar;
