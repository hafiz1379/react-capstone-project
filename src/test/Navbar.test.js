// import { useState, useEffect } from 'react';
// import { useNavigate, useLocation } from 'react-router-dom';

// const Navbar = () => {
//   const navigate = useNavigate();
//   const location = useLocation();
//   const [showButton, setShowButton] = useState(false);
//   const { pathname } = location;
//   let title;

//   useEffect(() => {
//     setShowButton(location.pathname !== '/');
//   }, [location.pathname]);

//   const handleGoBack = () => {
//     navigate(-1);
//   };
//   if (pathname === '/') {
//     title = 'Air Polution app';
//   } else if (pathname.startsWith('/') && pathname.split('/').length === 2) {
//     title = 'Stats';
//   }

//   return (
//     <header className="flex bg-blue_header p-2 justify-between">
//       { showButton && (
//       <button type="button" className="flex items-center text-xl" onClick={handleGoBack}>
//         <img src="https://api.iconify.design/ic:sharp-arrow-back-ios.svg?color=%23ffffff" alt="back arrow" />
//         2023
//       </button>
//       )}
//       <h1 className="font-normal">{title}</h1>
//       <div className="flex gap-3">
//         <img className="h-6" src="https://api.iconify.design/carbon:microphone-filled.svg?color=%23ffffff" alt="speak" />
//         <img className="h-6" src="https://api.iconify.design/clarity:settings-solid.svg?color=%23ffffff" alt="settings" />
//       </div>
//     </header>
//   );
// };
// export default Navbar;

import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Navbar', () => {
  test('renders title', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );

    const titleElement = screen.getByText(/Air Polution app/i);
    expect(titleElement).toBeInTheDocument();
  });

  test('renders correctly', () => {
    const { container } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly when pathname is /', () => {
    const { container } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly when pathname is /stats', () => {
    const { container } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
