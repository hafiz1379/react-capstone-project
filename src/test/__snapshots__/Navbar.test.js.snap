// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<header
  class="flex bg-blue_header p-2 justify-between"
>
  <h1
    class="font-normal"
  >
    Air Polution app
  </h1>
  <div
    class="flex gap-3"
  >
    <img
      alt="speak"
      class="h-6"
      src="https://api.iconify.design/carbon:microphone-filled.svg?color=%23ffffff"
    />
    <img
      alt="settings"
      class="h-6"
      src="https://api.iconify.design/clarity:settings-solid.svg?color=%23ffffff"
    />
  </div>
</header>
`;

exports[`Navbar renders correctly when pathname is / 1`] = `
<header
  class="flex bg-blue_header p-2 justify-between"
>
  <h1
    class="font-normal"
  >
    Air Polution app
  </h1>
  <div
    class="flex gap-3"
  >
    <img
      alt="speak"
      class="h-6"
      src="https://api.iconify.design/carbon:microphone-filled.svg?color=%23ffffff"
    />
    <img
      alt="settings"
      class="h-6"
      src="https://api.iconify.design/clarity:settings-solid.svg?color=%23ffffff"
    />
  </div>
</header>
`;

exports[`Navbar renders correctly when pathname is /stats 1`] = `
<header
  class="flex bg-blue_header p-2 justify-between"
>
  <h1
    class="font-normal"
  >
    Air Polution app
  </h1>
  <div
    class="flex gap-3"
  >
    <img
      alt="speak"
      class="h-6"
      src="https://api.iconify.design/carbon:microphone-filled.svg?color=%23ffffff"
    />
    <img
      alt="settings"
      class="h-6"
      src="https://api.iconify.design/clarity:settings-solid.svg?color=%23ffffff"
    />
  </div>
</header>
`;
